### PR TITLE
Adding IsInTransaction API to datastore package

### PR DESCRIPTION
**Pull request by zjusbo**
Add IsInTransaction API to datastore package. It is a very useful API to detect context environment and avoid nested transaction.

### CODE REVIEW

**Comment by okdave** [comment]
@adamtanner @zombiezen

File: datastore/transaction.go [hunk 1]
```diff
@@ -70,6 +70,14 @@ func RunInTransaction(c context.Context, f func(tc context.Context) error, opts
 	return ErrConcurrentTransaction
 }
 
+// IsInTransaction indicates whether the current context is a transaction context or not.
+func IsInTransaction(c context.Context) bool {
+	if internal.transactionFromContext(c) != nil {
```
Comment (zombiezen) [inline datastore/transaction.go]: Simplify this to `return internal.transactionFromContext(c) != nil`

**Comment by adamtanner** [comment]
Do you have a concrete example of where this functionality would simplify your code or make it more correct? I'm wondering if maybe there's a better way to structure the code such that this isn't necessary?

**Comment by adams-sarah** [comment]
+1 to what @adamtanner said.
The code itself is fine, but I'd rather not add more exported declarations unless we have a concrete example of why this is necessary.

**Comment by broady** [comment]
Closing due to lack of response.

**Pull request closed without merging**